### DISCUSSION TITLE
Nuke Calcs and Naming Systems

### DISCUSSION THREAD

**Discussion by SunlitZelkova** [PygmalionOfCyprus/cmo-db-requests#5309] [Q&A] [answered]
Hello,

I'm planning to make some DB requests on North Korean missiles, and have some questions.

1. **How is the yield of nuclear weapons implemented from a gameplay POV?**

So what I mean by this is if there is a degree of acceptable inaccuracy that DB unit modelers allow themselves, because from a gameplay POV the exact yield of a nuclear weapon doesn't matter. Nukes are usually used in game for targeting an air base or a marker (city). Because of this, giving a missile a 1 megaton warhead instead of a 300 kiloton one doesn't make much of a difference because either are more than enough to destroy even the largest international airport.

So, for example, it doesn't make sense to request that the KN-23 [Facility #3367] have its warhead changed from the 2.5 megaton one it has now to the standard 10 kiloton one used on other North Korean SRBMs and MRBMs, despite a 2.5 megaton warhead being quite literally 25 times larger than any known North Korean nuclear weapon's yield. It's a pointless request when 10 kilotons is going to level Osan Air Base anyways, i.e. the same gameplay effect as the 2.5 megaton warhead.

However, North Korea's new "tactical" nuclear warhead, the Hwasan-31, has a possible yield of just 900 tons- roughly on par with the warhead of an AIR-2 Genie rocket. This estimate comes from a statement from Kim Yo Jong, but it does match the dimensions of the warhead, which are similar to that of an AIR-2 warhead (WN-25, 44cm diameter, Hwasan-31's diameter is estimated to be about 46cm).

It would be very important to make sure the yield is modeled correctly, because such a low yield makes North Korean tactical nuclear weapons explicitly useless to a certain extent. If such a missile fell somewhere on Osan Air Base, it might take out 5-6 hardened aircraft shelters off to the side, but otherwise won't effect operations at the rest of the base.

Currently, the lowest yield nuclear weapon in the database is a 1 kiloton Soviet gravity bomb. Would it be possible to model sub-kiloton yield weapons in the game? If not, rounding up to 1 kiloton would not drastically affect gameplay.

I ask this question because when I requested the addition of the R-5M ballistic missile in the CWDB a couple years ago, it got added with the R-12's 1 megaton warhead instead of the 300 kiloton warhead specified in the request. That's fine for big weapons like that, but given how the North Korean SRBM warhead escalated from 10 kilotons to 2.5 megatons, I wanted to know more about how the decision is made to implement yield before requesting various missiles with the Hwasan-31.

2. **Is it better to request a nuclear missile as a weapon or a facility?**

There are discrepancies in how nuclear missiles are implemented in the DB. Some ballistic missiles are modeled as separate facilities, for example, there is a separate unit for the DF-26 Battalion (Nuclear) and DF-26 Battalion (Conventional). On there other hand, there are facilities that are modeled as a single unit but with different weapons that must be manually loaded in the weapon editor interface or via Lua. The KN-23 is an example of this.

I'm curious if there is a logic to this or some preferred way of requesting new missiles. The Hwasan-31 is designed to be mounted on the KN-23, KN-24, and KN-25. The latter two are already present in DB3000, but with conventional warheads only.

Would it be better to request nuclear variants of these missiles as optional weapons to be added on the existing facility/weapon mount, or request an entirely new facility?

3. **Are renaming requests acceptable?**

I ask because the naming system for North Korean missiles is all over the place right now and it makes it difficult to discern what is what, what is in service, etc. As an example, there are currently two differing entries for a single missile- the Hwasong-7. The Hwasong-7 is known to US intelligence as the Rodong (or Nodong). The Hwasong-7 entry in the DB is currently just a Scud-B battalion, while the Nodong-1 is an accurate representation of the missile. There is also a Nodong-2, which never existed. The nuclear variant is correctly marked as hypothetical but the conventional one should be too. Some missiles also exist that have a Korean name but no KN-XX designation.

If it's reasonable, I would create a request for a new naming system closer to the one currently used for Chinese missiles; Official Name [KN-XX], followed by any description in the comments section. This of course wouldn't be an urgent request, but something to help players navigate the different missiles in the long term. Missiles that lack a KN designation could simply be listed as "KN-XX," similar to how Chinese systems with an unknown "Type" designation are named in the DB.

**Comment by PygmalionOfCyprus** [comment, marked as answer]
1. Yield is implemented in game. We actually have a pretty complex nuke sim under the hood, though you're correct that when targeting single-unit facilities it doesn't really matter much. You'd notice it much more if using, say, BOL to target formations of ground units or hitting more complex facilities.

2. In general just request the weapon and we'll handle adding it to appropriate facilities. The Korean stuff hasn't been reviewed so I'm sure there's a lot of jank in the way the facilities are modelled. You can look at the Chinese PLARF units to see how we generally try and model stuff.

3. Absolutely. We're more than happy to have you help untangle the mess.
